### PR TITLE
Few link fixes in java doc & Added test suite for unit tests & Fixed few tests. 

### PR DESCRIPTION
**Pull request by ashutoshcp**
1. Fixed link to classes @link  <= @code
2. Added Suites for tests, Easy to run the tests in suite rather than running one by one
3. Fixed few tests

### CODE REVIEW

**Comment by craigatgoogle** [comment]
Thanks very much for the contribution! Much appreciated :)

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 language: java
 sudo: false
-script: mvn clean test
+script: mvn clean test -DrunSuite=**/GoogleOAuthPlugInTestSuite.class
```
Comment (craigatgoogle) [inline .travis.yml]: Out of curiosity, what advantage do you see of specifying a suite vs. allowing all unit tests to be run?

File: src/test/java/com/google/jenkins/GoogleOAuthPlugInTestSuite.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.google.jenkins;
+
+import com.google.jenkins.plugins.CredentialsOAuthTestSuite;
+import com.google.jenkins.plugins.UtilTestSuite;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+
+@RunWith(Suite.class)
+@Suite.SuiteClasses(value = {CredentialsOAuthTestSuite.class, UtilTestSuite.class})
+public class GoogleOAuthPlugInTestSuite {}
```
Comment (craigatgoogle) [inline src/test/java/com/google/jenkins/GoogleOAuthPlugInTestSuite.java]: s/GoogleOAuthPlugInTestSuite/GoogleOAuthPluginTestSuite

File: .travis.yml (hunk 1 above)
Comment (craigatgoogle) [reply .travis.yml]: Would prefer to configure the plugin suites which get executable via a pom file configuration, see: https://stackoverflow.com/questions/11762801/run-junit-suite-using-maven-command

Comment (ashutoshcp) [reply .travis.yml]: If some of the tests are in Beta phase, one shouldn't add them to suite and thus, No failure will happen. 
Apart from that you can run all tests in single suite at a time.
E.g: CredentialsOAuthTestSuite will run all tests inside: com.google.jenkins.plugins.credentials.oauth
and UtilTestSuite will run all tests inside com.google.jenkins.plugins.util package.
Moreover GoogleOAuthPlugInTestSuite is collection of all test suites,that is equivalent to running mvn test.

Comment (craigatgoogle) [reply .travis.yml]: That makes sense. Thanks!

File: src/test/java/com/google/jenkins/GoogleOAuthPlugInTestSuite.java (hunk 2 above)
Comment (craigatgoogle) [reply src/test/java/com/google/jenkins/GoogleOAuthPlugInTestSuite.java]: To clarify, could you please rename this class to: GoogleOAuthPluginTestSuite. Thanks.

Comment (ashutoshcp) [reply src/test/java/com/google/jenkins/GoogleOAuthPlugInTestSuite.java]: Done

**Comment by ashutoshcp** [comment]
@craigatgoogle  requested changes are done. do please merge if you find this useful.

**Pull request merged**